### DISCUSSION TITLE
Determine compatibility of Mithril Signer and Cardano Node

### DISCUSSION THREAD

**Discussion by TrevorBenson** [input-output-hk/mithril#1606] [General]
Is there any file in the Mithril repo which contains a minimum Cardano version for compatibility? 

----

## TL;DR

I'm looking to automate the Cardano Community Guild Operator Mithil scripts to update the Mithril release binaries. I ran into what I believe was a version compatibility issue some months ago on a testnet where I had to update the node to run the latest signer, or at least the errors I observed at the time went away after updating the Cardano Node and CLI, etc.

Due to this, if possible, I'd like to first verify if the current Cardano Node/CLI versions are compatible with the latest Mithril release. I'd prefer to not suggest any update which could break the SPO's setup, or at minimum warn them it may require updating Cardano as well if they update the Mithril binaries.

Thanks in advance for any suggestions.

**Comment by jpraynaud** [comment]
Hi @TrevorBenson, thanks for creating this discussion. 

We could create a file at the root of the repository that gives minimum version of the Cardano supported by the Mithril signer.

We would have 2 options:
- Have a `Cardano.version` file with only one version that is tied to the tag used to identify (pre)-release and unstable versions (i.e. you would have to target the correct tag to retrieve the minimum version depending on the Cardano network you are setting up):
```bash
$ cat Cardano.version
8.7.3
```
- Have a `Cardano.json` file where all the versions are listed at once and that should always be accessed on the main branch:
```bash
$ cat Cardano.json
{
  "mainnet": "8.7.3",
  "preprod": "8.9.1",
  "preview": "8.9.1",
  "sanchonet": "8.9.1"
}

```

**Comment by TrevorBenson** [comment]
@jpraynaud This sounds excellent.

The `Cardano.json` provides the most coverage. The Mithril scripts have which network stored in `mithril.env`. This should make it quite simple to leverage the network variable to find the latest release or prerelease binary based on network. Then leverage the new `Cardano.json` to confirm compatibility and decide how to inform the SPO.

**Reply by jpraynaud** [reply]
Hi @TrevorBenson, here is the issue created to handle the compatibility of Cardano node with Mithril signer: #1615 :+1:
